File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> numbers = new HashSet<>();
        for (int i : nums) {
            if (!numbers.contains(i)) numbers.add(i);
            else {return true; } // dup found
        }
        return false;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0 , len(nums) - 1
        res = nums[r]
        while l <= r:
            res = min(nums[l], nums[r], res)
            l += 1
            r -= 1
        return res